drift/lib/src/runtime/query_builder/expressions/case_when.dart

```diff
@@ -1,30 +1,56 @@
-import 'package:meta/meta.dart';
-
 import '../query_builder.dart';
 
+/// A `CASE WHEN` expression (without base expression) in sqlite.
+class CaseWhenExpression<R extends Object> extends _CaseWhenExpression<R> {
+  /// Creates a `CASE WHEN` expression from the independent components.
+  CaseWhenExpression({
+    required Map<Expression, Expression<R>> when,
+    Expression<R>? orElse,
+  }) : super._(null, when.entries.toList(), orElse);
+}
+
+/// A `CASE WHEN` expression (with base expression) in sqlite.
+class CaseWhenExpressionWithBase<T extends Object, R extends Object>
+    extends _CaseWhenExpression<R> {
+  /// Creates a `CASE WHEN` expression from the independent components.
+  CaseWhenExpressionWithBase(
+    Expression<T> base, {
+    required Map<Expression<T>, Expression<R>> when,
+    Expression<R>? orElse,
+  }) : super._(base, when.entries.toList(), orElse);
+}
```

Comment:
Separate classes are needed because of the type system.
For variant WITHOUT base - we do not care about the type of `when` expression (or maybe explicitly specify it to `Expression<bool>`).
For variant WITH base - `when` expressions type must match `base`'s type.